tests/drivers/bristlemouth/src/main.c

```diff
@@ -0,0 +1,146 @@
+/* main.c - OpenThread NCP */
+
+/*
+ * Copyright (c) 2020 Tridonic GmbH & Co KG
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+
+#include <device.h>
+#include <devicetree.h>
+
+#include <logging/log.h>
+#include <drivers/console/bm_serial.h>
+#include <drivers/bm_manchester/bm_manchester.h>
+
+#define LOG_MODULE_NAME bm_test
+LOG_MODULE_REGISTER(LOG_MODULE_NAME, LOG_LEVEL_DBG);
+
+/* Basic test with 2 byte payload before delimiter */
+static int basic_test_0 (void)
+{
+    int i;
+    bm_test_ret_t ret;
+    int success = 0;
+    uint8_t test_buf[8] = { 0xAA, 0xAA, 0XAA, 0xAA, 0xFF, 0xFF, 0xAA, 0xAA};
+    uint8_t expected_out[4] = {0xFF, 0xFF, 0xAA, 0xAA};
+
+    for (i = 0; i < sizeof(test_buf); i++)
+    {
+        ret = bm_serial_process_byte(test_buf[i]);
+    }
+
+    if (ret.retval == 0 && ret.length == 4)
+    {
+        for (i = 0; i < sizeof(expected_out); i++)
+        {
+            if (expected_out[i] != ret.buf_ptr[i])
+            {
+                goto out;
```

Comment:
I recommend avoiding gotos for things like these. It's very easy to screw things up later on once the code gets more complex.

In this case, I'd just use `break;` to get out of the loop since that's basically what you're doing.
You'll have to set success a different way, but it will be clearer. You can set success to 1 in line 39 before the for loop and set it to 0 right before you break to have the same effect.